### ISSUE TITLE
Valid Number

### ISSUE THREAD

**Issue by ziyeqinghan** [ziyeqinghan/leetcode#46]
In [discuss 1](https://discuss.leetcode.com/topic/57428/share-my-answer-about-valid-number-3ms-6ms-i-think-it-s-a-word-game-somehow/2) :
Q: what's valid number in this issue?
A: float(+'e'+integer).
Q: what's valid float number?
A: such as `12.3` `12.` `.3` and integer included.
Q: when `+` or `-` can be showed?
A: We can only see `+` and `-` in the beginning and after an `e`.

In [discuss 2](https://discuss.leetcode.com/topic/9490/clear-java-solution-with-ifs/2) :
So, basically the number should match this regular expression:
`[-+]?(([0-9]+(.[0-9]*)?)|.[0-9]+)(e[-+]?[0-9]+)?`
